config/service_config.go

```diff
@@ -105,6 +106,45 @@ func NewServiceConfig(id string, context context.Context) *ServiceConfig {
 	}
 }
 
+// Get Random Ports with no size
+func getRandomPort() int {
+	return getRandomPorts(1)[0]
+}
+
+// Get Random Ports with size
+func getRandomPorts(size int) []int {
+	ports := make([]int, 0, size)
+	if size <= 0 {
+		return ports
+	}
+
+	var (
+		flag bool
+		addr *net.TCPAddr
+	)
+	i := 0
+	for i < size {
+		flag = false
+		addr = nil
+		go func() {
+			listener, err := net.Listen("tcp", ":0")
```

Comment:
Whether it is necessary to use the goroutine here, because one goroutine block and wait for 50 milliseconds, then 10 addrs will take at least 500 milliseconds. Personally, I think it can be dealt with in the for loop.

Reply:
agree with u.

Reply:
agree